src/test/java/me/karakelley/http/HttpServerTest.java

```diff
@@ -0,0 +1,169 @@
+package me.karakelley.http;
+
+import me.karakelley.http.helpers.ClientHelper;
+import me.karakelley.http.mocks.MockLogger;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.net.ConnectException;
+import java.net.NoRouteToHostException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class HttpServerTest {
+  private ClientHelper client;
+  private HttpServer httpServer;
+  private HttpServer httpServerOnSamePort;
+  private ServerLogger mockLogger = new MockLogger("Servertest");
+  int count;
+  int maxTries;
+
+  @BeforeEach
+  void setUp() {
+    client = new ClientHelper();
+    count = 0;
+    maxTries = 4;
+  }
+
+  @Test
+  void testSendsResponseGivenValidRequest() throws Exception {
+    httpServer = new HttpServer(0, mockLogger);
+    ExecutorService executorService = Executors.newSingleThreadExecutor();
+    executorService.submit(() -> httpServer.start());
+
+    while (true) {
```

Comment:
Lots of duplication in this test.

Reply:
In regards to initializing the `httpServer` then starting it two lines down?